source/javascripts/gradient_fade.js

```diff
@@ -0,0 +1,16 @@
+function init() {
+    // Create the listener function
+    var updateLayout = _.debounce(function(e) {
```

Comment:
Yeah, agree. At the very least you could use waypoints to handle this, would be easier than calculating distance and probably more performative.